specs/validator/0_beacon-chain-validator.md

```diff
@@ -214,13 +214,21 @@ Set `block.slot = slot` where `slot` is the current slot at which the validator
 
 ##### Parent root
 
-Set `block.parent_root = signing_root(parent)`.
+Set `block.parent_root = hash_tree_root(parent)`.
 
 ##### State root
 
 Set `block.state_root = hash_tree_root(state)` of the resulting `state` of the `parent -> block` state transition.
 
-*Note*: To calculate `state_root`, the validator should first run the state transition function on an unsigned `block` containing a stub for the `state_root`. It is useful to be able to run a state transition function that does _not_ validate signatures or state root for this purpose.
+*Note*: To calculate `state_root`, the validator should first run the state transition function on an unsigned `block` containing a stub for the `state_root`.
+It is useful to be able to run a state transition function that does _not_ validate signatures or state root for this purpose:
+
+```python
+def compute_new_state_root(state: BeaconState, block: BeaconBlock) -> Root:
+    process_slots(state, block.slot)
```

Comment:
shouldn't we instead use `state_transition` with the `validate_results` set to false here?

Also kind of sucks that these functions all mutate `state`, but that's not changing

Reply:
`state_transition` uses a signed block as input, which makes no sense for a proposer that is in the process of producing it. This is the kind of thing Nimbus is struggling with, empty/placeholder signatures are bad. Alternatively we can make `state_transition` go on the unsinged block, and make signature verification with proposer a separate thing, but prefer not to go that far. Regardless, I think this `compute_new_state_root` is so minimal (just call two functions cleanly, without any placeholder info or transformed data), that it is a good design to promote as spec.